### ISSUE TITLE
关于BPNP损失是否参与参数更新问题

### ISSUE THREAD

**Issue by 10qqq** [EasonChen99/I2D-Loc#8]
BPNP loss 的梯度链被切断
在 main_bpnp.py (line 99)：
flow_up 先被 flow_up.int() 离散化；
之后 output.cpu().permute(...).numpy() 转 NumPy；
再用 torch.tensor(...) 重新构造 pts2d/pts3d。
这条路径已经和原始 flow_preds[-1] 脱离计算图。即使后面调用：
main_bpnp.py (line 122)
python
P_out = bpnp(pts2d, pts3d, K)
由于 pts2d/pts3d/K 都不是带计算图的输入，BPnP.backward() 实际不会把梯度传回网络预测的 flow。
requires_grad = True 放错位置，不能恢复断掉的图
main_bpnp.py (line 134)
python
pts2d_gt.requires_grad = True
pts2d_pro.requires_grad = True
这只会让 pts2d_gt/pts2d_pro 自己成为梯度叶子，不会重新连接到 P_out -> BPnP -> pts2d -> flow_up，更不会连接到 RAFT 参数。

**Comment by EasonChen99** [comment]
Thanks for the careful observation. Yes, you are right. I will correct this in the uploaded version later.

**Comment by 10qqq** [comment]
改动后的4096个点是测试过的吗？还是一个参考？

**Comment by EasonChen99** [comment]
It is mainly for reference. I will test it after I finish my current tasks and update the repository.
